source/game/kart/KartPullPath.cc

```diff
@@ -0,0 +1,286 @@
+#include "KartPullPath.hh"
+
+#include "game/system/CourseMap.hh"
+
+#include <egg/geom/Plane.hh>
+
+namespace Kart {
+
+/// @addr{0x8059308C}
+KartPullPathTracker::KartPullPathTracker(KartPullPath *handle, Type type)
+    : m_type(type), m_currentIdx(0), m_pointInfo(nullptr), m_handle(handle) {}
+
+/// @addr{0x805930E4}
+KartPullPathTracker::~KartPullPathTracker() = default;
+
+void KartPullPathTracker::calc() {
+    ASSERT(m_pointInfo);
+
+    if (m_type == Type::Global) {
+        calcTrackerGlobal();
+    } else if (m_type == Type::Regional) {
+        calcTrackerRegional();
+    }
+}
+
+/// @addr{0x80593138}
+void KartPullPathTracker::calcTrackerGlobal() {
+    s16 idx;
+    EGG::Vector3f point, dir;
+    if (search(SearchDirection::Current, idx, point, dir)) {
+        m_handle->changePoint(idx, getDistance(point, dir));
+    }
+
+    if (static_cast<size_t>(++m_currentIdx) >= m_pointInfo->pointCount()) {
```

Comment:
This causes a desync on the KC ng wr
```suggestion
    if (static_cast<size_t>(++m_currentIdx) >= m_pointInfo->pointCount() - 1) {
```